Tests/BotTests/GitHub/GitHubEventsTests.swift

```diff
@@ -69,27 +89,49 @@ class GitHubEventsTests: XCTestCase {
     }
 
     func test_handling_unknown_event() {
-        let service = GitHubService()
+        let service = GitHubService(signatureToken: signatureToken)
+
+        let payload = "{}".data(using: .utf8)!
 
         let request = StubbedRequest(
-            headers: [GitHubService.HTTPHeader.event.rawValue : "anything_really"],
-            body: nil
+            headers: [
+                GitHubService.HTTPHeader.event.rawValue : "anything_really",
+                GitHubService.HTTPHeader.signature.rawValue: signature(for: payload)
+            ],
+            data: payload
         )
 
         let result = service.handleEvent(from: request).first()
 
         expect(result?.error) == .unknown
     }
+
+    func test_handling_untrustworthy_payload() {
+        let service = GitHubService(signatureToken: signatureToken)
+
+        let request = StubbedRequest(
+            headers: [GitHubService.HTTPHeader.event.rawValue : "anything_really"],
+            data: nil
+        )
+
+        let result = service.handleEvent(from: request).first()
+
+        expect(result?.error) == .untrustworthy
+    }
```

Comment:
This one mainly test if a request _doesn't have_ the expected signature header at all, but we should also have a test with the header present but an invalid value (no `sha=` prefix) and another one with an invalid signature (`sha=someWellFormedSHAButNotTheExpectedSignature`)

Reply:
You're right, better be safe 👍